pkg/neo/executor/executor.go

```diff
@@ -0,0 +1,29 @@
+package executor
+
+import (
+	"github.com/casbin-mesh/neo/pkg/db"
+	"github.com/casbin-mesh/neo/pkg/neo/session"
+	"github.com/casbin-mesh/neo/pkg/primitive"
+	"github.com/casbin-mesh/neo/pkg/primitive/btuple"
+)
+
+type Executor interface {
+	Init()
+	Next(tuple *btuple.Reader, rid *primitive.ObjectID) (bool, error)
```

Comment:
The immutable reader may be costly to modify. Change the Reader to Modifier?  e.g., the update executor; we need to modify the tuples generated from the children's executors. And the parent executor should take ownership of tuples from the children's executors.